tools/webpack/packages.js

```diff
@@ -146,7 +146,7 @@ module.exports = {
 	},
 	plugins: [
 		...plugins,
-		new DependencyExtractionWebpackPlugin( { injectPolyfill: true } ),
+		new DependencyExtractionWebpackPlugin( { injectPolyfill: false } ),
```

Comment:
What do you think of making this configurable?

I understand this would require more setup to work correctly for third parties. 

This would behave like `false` and apply the new behavior of searching for the comment string.

```suggestion
		new DependencyExtractionWebpackPlugin( { injectPolyfill: 'auto' } ),
```

Maybe this isn't necessary, but it would also avoid applying this for third parties when it's not desired or add a path to expose it to third parties in the future.

Reply:
I'm happy to make changes, but I think we need to clarify the semantics here. As far as I can tell, what you're looking for is:

- `false`: Don't add polyfills, regardless of the presence of magic comments.
- `true`: Add polyfills, regardless of the presence of magic comments.
- `'auto'`: Add polyfills only when magic comments are present.

Does that sounds about right?

To clarify, note that this is a separate build step to the one that actually adds the magic comments.
`node ./bin/packages/build.js` builds the packages in situ, and `wp-scripts build` is the one that takes each built package from its src directory, extracts the dependencies, and places them in the top-level `build` dir.

My reasoning for not adding a separate option was that if we didn't want auto polyfilling we probably wouldn't add magic comments in the first build step anyway (the `addPolyfillComments` option), but I'm happy to make changes if we feel it's still best to make this behaviour explicitly opt-in in DEWP as well.

Reply:
I'm not 100% decided on this… let's leave it as-is.

Reply:
The strategy outlined seems good overall. The question is whether we really want to keep the differences between false and auto? The magic comment gets injected only with custom Babe code. In effect, in WP ecosystem there should never be a case where polyfill gets listed as a dependency outside of Gutenberg and Core. The only exception would be a project that bundles in a custom  way WP packages but it's unlikely they would use the webpack plugin to generate asset files.

Reply:
That said, should we ever enable magic comments by default in the default Babel preset that WordPress maintains?

Reply:
> In effect, in WP ecosystem there should never be a case where polyfill gets listed as a dependency outside of Gutenberg and Core

Someone already filed https://github.com/Automattic/jetpack/issues/39452 asking us to do it in Jetpack. 😀

I can't see any reason anyone would have the magic comment and not want the plugin to add the dependency.

Reply:
It looks like a follow-up is very welcomed 👌